course/src/main/java/org/openedx/course/presentation/container/CourseContainerFragment.kt

```diff
@@ -352,21 +359,51 @@ fun CourseDashboard(
                     courseImage = courseImage,
                     imageHeight = 200,
                     expandedTop = {
-                        ExpandedHeaderContent(
-                            courseTitle = viewModel.courseName,
-                            org = viewModel.organization
-                        )
+                        if (dataReady.value == true) {
+                            ExpandedHeaderContent(
+                                courseTitle = viewModel.courseName,
+                                org = viewModel.courseStructure?.org!!
+                            )
+                        }
                     },
                     collapsedTop = {
                         CollapsedHeaderContent(
                             courseTitle = viewModel.courseName
                         )
                     },
+                    upgradeButton = {
+                        if (dataReady.value == true && canShowUpgradeButton) {
+                            val horizontalPadding = if (!windowSize.isTablet) 16.dp else 98.dp
+                            UpgradeToAccessView(
+                                modifier = Modifier.padding(
+                                    start = horizontalPadding,
+                                    end = 16.dp,
+                                    top = 16.dp
+                                ),
+                                type = UpgradeToAccessViewType.COURSE,
+                            ) {
+                                IAPDialogFragment.newInstance(
+                                    iapFlow = IAPFlow.USER_INITIATED,
+                                    screenName = IAPAnalyticsScreen.COURSE_DASHBOARD.screenName,
+                                    courseId = viewModel.courseId,
+                                    courseName = viewModel.courseName,
+                                    isSelfPaced = viewModel.courseStructure?.isSelfPaced!!,
+                                    productInfo = viewModel.courseStructure?.productInfo!!
+                                ).show(
+                                    fragmentManager,
+                                    IAPDialogFragment::class.simpleName
```

Comment:
to prevent issues after obfuscation, please make a 
`public const val TAG = "IAPDialogFragment"` inside IAPDialogFragment and use it here

Reply:
added.